src/Components/CreateImageWizard/utilities/requestMapper.ts

```diff
@@ -311,8 +318,34 @@ export const mapExportRequestToState = (
       parent_id: request.metadata?.parent_id || null,
       exported_at: request.metadata?.exported_at || '',
     },
-    env: initialState.env,
-    registration: initialState.registration,
+    env: subscribed
+      ? {
+          serverUrl:
+            request.customizations.subscription?.['server-url'] ||
+            initialState.env.serverUrl,
+          baseUrl:
+            request.customizations.subscription?.['base-url'] ||
+            initialState.env.baseUrl,
+        }
+      : {
+          serverUrl: '',
+          baseUrl: '',
+        },
+    registration: subscribed
+      ? {
+          registrationType: request.customizations?.subscription
+            ? request.customizations.subscription.rhc
+              ? 'register-now-rhc'
+              : 'register-now-insights'
+            : initialState.registration.registrationType,
+          activationKey:
+            request.customizations.subscription?.['activation-key'] ||
+            initialState.registration.activationKey,
+        }
+      : {
+          registrationType: 'register-later',
+          activationKey: '',
```

Comment:
Just to make this more structure and readable, we could first create a `subscriptionState` object, using the `subscribed` flag just once.

```ts
  const subscriptionState = subscribed
    ? {
        serverUrl: request.customizations.subscription?.['server-url'] || initialState.env.serverUrl,
        baseUrl: request.customizations.subscription?.['base-url'] || initialState.env.baseUrl,
        registrationType: request.customizations.subscription
          ? request.customizations.subscription.rhc
            ? 'register-now-rhc'
            : 'register-now-insights'
          : initialState.registration.registrationType,
        activationKey: request.customizations.subscription?.['activation-key'] || initialState.registration.activationKey,
      }
    : {
        serverUrl: '',
        baseUrl: '',
        registrationType: 'register-later' as const,
        activationKey: '',
      };
```

and return 
```ts
  return {

    // ...metadata...

    env: {
      serverUrl: subscriptionState.serverUrl,
      baseUrl: subscriptionState.baseUrl,
    },
    registration: {
      registrationType: subscriptionState.registrationType,
      activationKey: subscriptionState.activationKey,
    },
    ...commonRequestToState(blueprintResponse),
```